utils/debugging.py

```diff
@@ -0,0 +1,82 @@
+from pprint import pprint
+from functools import reduce
+
+"""
+Shorten values of a dict to a maximum length
+
+Args:
+    body (dict): the dict to be shortened
+    maxlen (int): maximum length to shorten the fields to
+
+Returns:
+    dict: the dict with its values shortened
+"""
+shorten_all = (
+    lambda body, maxlen=1024: {k: shorten_all(v) for k, v in body.items()}
+    if type(body) is dict
+    else list(map(shorten_all, body))
+    if type(body) is list
+    else f"{body[:maxlen-3]}..."
+    if type(body) is str and len(body) > maxlen
+    else body
+)
+
+def shprint(obj: dict, maxlen: int=1024):
+    """
+    Pretty-print a dict, keeping all the values shortened to a
+    maximum length, so that they don't dominate the visual output
+
+    Args:
+        obj (dict): the dict to be printed
+        maxlen (int): maximum character length of a dict's fields
+    """
+    pprint(shorten_all(obj, maxlen))
+
+def bigprint(msg: str, margin: int=1, **kwargs):
+    """
+    Print a big string padded with a box :)
+
+    Args:
+        msg (str): the message to be padded and printed
+        margin (int): size of the whitespace inside the box
+        **kwargs: all additional kwargs are passed directly to boxpad(),
+            refer to the doc for that function for additional info
+    """
+    print(boxpad(boxpad(msg, **{**kwargs, "char":' ', "padding":margin}), **{**kwargs, "char":'#'}))
+
+def beeeeeegprint(msg: str, layers: int=3, **kwargs):
```

Comment:
can you call this `biggerprint` or `xlprint`, less likely to misspell?

Reply:
kk......